src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -11,113 +11,342 @@
+            {
+                case Android.Locations.LocationManager.NetworkProvider:
+                    posSource = Windows.Devices.Geolocation.PositionSource.Cellular;    // cell, wifi
+                    break;
+                case Android.Locations.LocationManager.PassiveProvider:
+                    posSource = Windows.Devices.Geolocation.PositionSource.Unknown;  // other apps
+                    break;
+                case Android.Locations.LocationManager.GpsProvider:
+                    posSource = Windows.Devices.Geolocation.PositionSource.Satellite;
+                    break;
+                default:
+                    // ex.: "fused" - all merged, also e.g. Google Play
+                    posSource = Windows.Devices.Geolocation.PositionSource.Unknown;
+                    break;
+            }
+
+            Windows.Devices.Geolocation.BasicGeoposition basicGeoposition; 
+            basicGeoposition.Altitude = location.Altitude;
+            basicGeoposition.Latitude = location.Latitude;
+            basicGeoposition.Longitude = location.Longitude;
+
+            Windows.Devices.Geolocation.Geopoint geopoint = new Windows.Devices.Geolocation.Geopoint(basicGeoposition,
+                        Windows.Devices.Geolocation.AltitudeReferenceSystem.Ellipsoid,
+                        Wgs84SpatialReferenceId
+                    );
+
+            double? locVertAccuracy = null;
+            // VerticalAccuracy is since API 26
+            if (Android.OS.Build.VERSION.SdkInt >= Android.OS.BuildVersionCodes.O)
+            {
+                if (location.HasVerticalAccuracy)
+                {
+                    locVertAccuracy = location.VerticalAccuracyMeters;
+                }
+            }
+
+
+            Windows.Devices.Geolocation.Geoposition geopos = new Windows.Devices.Geolocation.Geoposition(
+                new Windows.Devices.Geolocation.Geocoordinate(
+                    latitude: location.Latitude,
+                    longitude: location.Longitude,
+                    altitude: location.Altitude,
+                    timestamp: FromUnixTime(location.Time),
+                    speed: location.HasSpeed ? location.Speed : 0,
+                    point: geopoint,
+                    accuracy: location.HasAccuracy ? location.Accuracy : 0,
+                    altitudeAccuracy: locVertAccuracy,
+                    heading: geoheading,
+                    positionSource: posSource
+                )
+            );
+
+
+
+            // this code doesn't set:
+            // heading (double?) ,
+            // position source, [enum]
+            // position sourcetimestamp (DateTimeOffset?),
+            // satelitedata
+
+
+            return geopos;
+        }
+
+
+        private static DateTimeOffset FromUnixTime(long time)
+        {
+            var epoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+            return epoch.AddMilliseconds(time);
+        }
```

Comment:
`DateTimeOffset.FromUnixTimeMilliseconds()`

Reply:
I didn't know about this method, so I didn't change this method.
If this is so short, I removed FromUnixTime helper method completely.